### DISCUSSION TITLE
Nuxt3 and OpenTelemetry

### DISCUSSION THREAD

**Discussion by BiffBaffBoff** [nuxt/nuxt#16498] [Questions]
Hi Everyone,

I've started learning Nuxt over the past week and something I'm always interested in is gathering telemetry information and exporting that to something like honeycomb. I've followed their guide here: https://docs.honeycomb.io/getting-data-in/opentelemetry/javascript/ And was able to get my server up and running via docker but no telemetry information was being gathered.

I had a look on the nuxt and nitro website but was unable to find any information about open telemetry.

For reference this is part of my dockerfile:

```
FROM node:18-alpine AS runner
WORKDIR /app

COPY --from=installer /app/apps/web/ .
RUN yarn install #this is to install the OpenTelemetry packages referenced in package.json and used in ./tracing.js

ENV NUXT_HOST=0.0.0.0
ENV NUXT_PORT=8080
ENV PORT=8080
ENV OTEL_EXPORTER_OTLP_ENDPOINT="https://api.honeycomb.io"
ENV OTEL_SERVICE_NAME="nuxt-testing"

EXPOSE 8080

CMD node -r ./tracing.js .output/server/index.mjs
```

The Api key env variable is set on my deployment server and injected at runtime so its not part of my image for security purposes.
`OTEL_EXPORTER_OTLP_HEADERS="x-honeycomb-team=your-api-key"`

Any help or ideas would be appreciated, thanks.

**Comment by JPeer264** [comment]
Hey, your example should actually work as you require the `./tracing.js`. There could be a chance that the OTLP exporter don't have the actual permissions to send the traces to honeycomb. Without adding any `OTEL_` environment variables you can run Jaeger locally and check if the exporter is correctly setup:

```yaml
# docker-compose service example
jaeger:
    image: jaegertracing/all-in-one:1.50.0
    environment:
      COLLECTOR_OTLP_ENABLED: 'true'
    ports:
    - 16686:16686
    - 14250:14250
    - 4318:4318
```

If you **don't see** your service within Jaeger, then there might be a problem with your setup of the otel instrumentation.
If you **do see** your service within Jaeger, then the exporter doesn't have the proper credentials to send traces to honeycomb.

Make sure in `./tracing.js` you have the otlp exporter enabled:

```js
new opentelemetry.NodeSDK({
  ...
  traceExporter: new OTLPTraceExporter(),
  ...
});
```

**Reply by hugobytebender** [reply]
does nuxt3 still uses nuxt/telemetry or does it use something else ?

**Reply by JPeer264** [reply]
FWIW `@nuxt/telemetry` has nothing to do with OpenTelemetry, as it is only for usage analytics for Nuxt itself.
